### PR TITLE
Restore support of Shelley 2-tuple tx body

### LINKED ISSUE #3224: `cardano-cli transaction view` deserialise error
Passing a signed tx looks great, but passing a txbody throws a deserialise error:

```
cardano-cli transaction view --tx-body-file shelley/tx-proposal-d-0.txbody
Command failed: transaction view  Error: shelley/tx-proposal-d-0.txbody: TextEnvelope decode error: DecoderErrorDeserialiseFailure "Shelley TxBody" (DeserialiseFailure 1 "expected tx body tuple of size 3, 4 or 6")
```

_Originally posted by @disassembler in https://github.com/input-output-hk/cardano-node/issues/3208#issuecomment-921069152_

### PR DESCRIPTION
**Pull request by cblp**
Restore support of 2-tuple tx body (fix #3224)

### CODE REVIEW

**Review by JaredCorduan — APPROVED** [review]
Thanks @cblp

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 1]
```diff
@@ -1470,6 +1470,17 @@ deserialiseShelleyBasedTxBody era bs =
...
+          txbody     <- fromCBOR
+          txmetadata <- CBOR.decodeNullMaybe fromCBOR
+          pure $ CBOR.Annotator $ \fbs ->
+            ShelleyTxBody
+              era
+              (CBOR.runAnnotator txbody fbs)
+              [] -- scripts
+              (CBOR.runAnnotator (pure TxBodyNoScriptData) fbs)
+              (fmap (`CBOR.runAnnotator` fbs) txmetadata)
+              (CBOR.runAnnotator (pure TxScriptValidityNone) fbs)
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/TxBody.hs:1483]: If you're copying and pasting anyway, don't add extra unnecessary changes too. It makes it harder to compare to see that they're the same.
```suggestion
              (flip CBOR.runAnnotator fbs (return TxBodyNoScriptData))
              (fmap (flip CBOR.runAnnotator fbs) txmetadata)
              (flip CBOR.runAnnotator fbs (return TxScriptValidityNone))
```
If you think the existing code is poor style then change that too. Or share the code so there's only a single copy and make that the nice style. But don't copy with minor unnecessary changes, that way lies madness.

**Review by dcoutts — APPROVED** [review]
More specifically, it's not just pre-Alonzo, but support for unsigned tx bodies made by the CLI during the Shelley era _before_ we added CLI support for multi-sig scripts.

Please update the PR description. The current description confused people and made it sound like an urgent bug fix, which is far far from the reality.

File: cardano-api/src/Cardano/Api/TxBody.hs (hunk 1 above)
Comment (cblp) [reply cardano-api/src/Cardano/Api/TxBody.hs:1483]: Sorry, I had to simplify this to get into it.

Comment (cblp) [reply cardano-api/src/Cardano/Api/TxBody.hs]: done

**Review by dcoutts — APPROVED** [review]
LGTM

**Comment by cblp** [comment]
bors merge

**Pull request merged**